compile/service.go

```diff
@@ -207,6 +207,11 @@ func (f *FunctionSpec) Link(scope Scope) error {
 	return nil
 }
 
+// HasReturn returns true if this function has a non-void return.
+func (f *FunctionSpec) HasReturn() bool {
+	return f.ResultSpec != nil && f.ResultSpec.ReturnType != nil
```

Comment:
(unrelated to this PR), should we always have a `ResultSpec` for a function, even if it's void, and have the `ReturnType` be `nil`? I have a similar check in yab but it might be cleaner to assume `ResultSpec != nil`